### PR TITLE
some performance improvement

### PR DESCRIPTION
**Pull request by fanatid**
Benchmark:
old:
```
$ SEED=52d036450e91515f807900d2f2f73fc4194f3f0d602283f30509e41b4b5c472e node benchmark/benchmark.js
Seed: 52d036450e91515f807900d2f2f73fc4194f3f0d602283f30509e41b4b5c472e
--------------------------------------------------
encode x 51,640 ops/sec ±1.31% (9 runs sampled)
decode x 76,285 ops/sec ±0.45% (9 runs sampled)
==================================================
```

new:
```
$ SEED=52d036450e91515f807900d2f2f73fc4194f3f0d602283f30509e41b4b5c472e node benchmark/benchmark.js
Seed: 52d036450e91515f807900d2f2f73fc4194f3f0d602283f30509e41b4b5c472e
--------------------------------------------------
encode x 65,623 ops/sec ±1.06% (8 runs sampled)
decode x 287,033 ops/sec ±0.31% (9 runs sampled)
==================================================
```

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -16,76 +20,82 @@ module.exports = function base (ALPHABET) {
...
-
-      for (j = 0; j < bytes.length; j++) bytes[j] *= BASE
-      bytes[0] += ALPHABET_MAP[c]
-
-      var carry = 0
-      for (j = 0; j < bytes.length; ++j) {
-        bytes[j] += carry
+      var carry = bytes[0] * BASE + value
+      bytes[0] = carry & 0xff
+      carry >>= 8
```
Comment (dcousens) [inline index.js]: nice

**Comment by dcousens** [comment]
Awesome @fanatid :+1:

**Comment by fanatid** [comment]
Thanks! Publish please :)

File: package.json [hunk 2]
```diff
@@ -28,15 +14,31 @@
...
+    "standard": "standard",
+    "test": "npm run unit && npm run standard",
+    "unit": "mocha"
+  },
   "devDependencies": {
     "benchmark": "^2.0.0",
-    "mocha": "*",
-    "standard": "*"
+    "mocha": "^2.3.4",
+    "node-gyp": "^3.2.1",
```
Comment (dcousens) [inline package.json]: removed `node-gyp` in 485a693,  looks like you accidentally left it in

File: index.js [hunk 3]
```diff
@@ -16,76 +20,82 @@ module.exports = function base (ALPHABET) {
...
+      var value = ALPHABET_MAP[string[i]]
+      if (value === undefined) {
+        throw new Error('Non-base' + BASE + ' character')
+      }
 
-    for (i = 0; i < string.length; i++) {
-      var c = string[i]
-      if (!(c in ALPHABET_MAP)) throw new Error('Non-base' + BASE + ' character')
-
-      for (j = 0; j < bytes.length; j++) bytes[j] *= BASE
```
Comment (dcousens) [inline index.js]: @fanatid can you elaborate on why we were able to drop the loop here?
I'm going to read over this again before I release a version,  and I'm 99% sure I understand why :)

Comment (fanatid) [inline index.js]: I just moved multiplication to loop below

Comment (dcousens) [inline index.js]: Right,  of course.  Sorry...   Will review after sleep.

**Pull request merged**